autopilot/hostfilter.go

```diff
@@ -226,48 +226,48 @@ func isUsableContract(cfg api.AutopilotConfig, ci contractInfo, bh uint64, rente
 
 	if bh > c.EndHeight() {
 		reasons = append(reasons, errContractExpired.Error())
-		renew = false
-		refresh = false
+		usable = false
 		recoverable = false
+		refresh = false
+		renew = false
 	} else if c.Revision.RevisionNumber == math.MaxUint64 {
 		reasons = append(reasons, errContractMaxRevisionNumber.Error())
-		renew = false
-		refresh = false
+		usable = false
 		recoverable = false
+		refresh = false
+		renew = false
 	} else {
 		if isOutOfCollateral(c, s, renterFunds, bh) {
 			reasons = append(reasons, errContractOutOfCollateral.Error())
-			renew = false
-			refresh = true
+			usable = false
 			recoverable = true
+			refresh = true
+			renew = false
 		}
 		if isOutOfFunds(cfg, s, c) {
 			reasons = append(reasons, errContractOutOfFunds.Error())
-			renew = false
-			refresh = true
+			usable = false
 			recoverable = true
+			refresh = true
+			renew = false
 		}
 		if shouldRenew, secondHalf := isUpForRenewal(cfg, *c.Revision, bh); shouldRenew {
-			if secondHalf {
-				reasons = append(reasons, errContractUpForRenewal.Error()) // only unusable if in second half of renew window
-			}
-			renew = true
-			refresh = false
+			reasons = append(reasons, fmt.Errorf("%w; second half: %t", errContractUpForRenewal, secondHalf).Error())
```

Comment:
Adding a `reason` here in all cases helps with debugging and should get rid of potential "unknown" reasons for contracts that get removed from the set. I believe this is the only case when a contract can be deemed `unsable` and not have a reason logged. Added `second half` for extra context.